pkg/controller/drain/drain_controller.go

```diff
@@ -343,6 +304,60 @@ func (ctrl *Controller) syncNode(key string) error {
 	return nil
 }
 
+func (ctrl *Controller) drainNode(node *corev1.Node, drainer *drain.Helper) error {
+	isOngoingDrain := false
+	var duration time.Duration
+
+	for k, v := range ctrl.ongoingDrains {
+		if k != node.Name {
+			continue
+		}
+		isOngoingDrain = true
+		duration = time.Now().Sub(v)
+		glog.Infof("Previous node drain found. Drain has been going on for %v hours", duration.Hours())
+		if duration > drainTimeoutDuration {
+			// TODO right now the daemon will alert to match previous behaviour. Consider having controller do so instead.
+			glog.Errorf("node %s: drain exceeded timeout: %v. Will continue to retry.", node.Name, drainTimeoutDuration)
+			ctrlcommon.MCCDrainErr.Set(1)
+		}
+		break
+	}
+
+	if !isOngoingDrain {
+		ctrl.logNode(node, "cordoning")
+		// perform cordon
+		if err := ctrl.cordonOrUncordonNode(true, node, drainer); err != nil {
+			ctrlcommon.MCCDrainErr.Set(1)
+			return fmt.Errorf("node %s: failed to cordon: %w", node.Name, err)
+		}
+		ctrl.ongoingDrains[node.Name] = time.Now()
+	}
+
+	// Attempt drain
+	ctrl.logNode(node, "initiating drain")
+	if err := drain.RunNodeDrain(drainer, node.Name); err != nil {
+		// To mimic our old daemon logic, we should probably have a more nuanced backoff.
+		// However since the controller is processing all drains, it is less deterministic how soon the next drain will retry,
+		// Anywhere between instant (if a node change happened) or up to hours (if there are many nodes competing for resources)
+		// For now, let's say if a node has been trying for a set amount of time, we make it less prioritized.
+		if duration > drainRequeueFailingThreshold {
+			ctrl.logNode(node, "Drain failed. Drain has been failing for more than %v minutes. Waiting %v minutes then retrying. "+
+				"Error message from drain: %v", drainRequeueFailingThreshold.Minutes(), drainRequeueFailingDelay.Minutes(), err)
+			ctrl.enqueueAfter(node, drainRequeueFailingDelay)
+		} else {
+			ctrl.logNode(node, "Drain failed. Waiting %v minute then retrying. Error message from drain: %v",
+				drainRequeueDelay.Minutes(), err)
+			ctrl.enqueueAfter(node, drainRequeueDelay)
+		}
+		ctrlcommon.MCCDrainErr.Set(1)
```

Comment:
Do we need to set this here after each failure? I thought we'd only want to alert after the 1hr mark like before (as done above)

Reply:
I took a quick look at [how it was before](https://github.com/openshift/machine-config-operator/blob/6e5ce0b171407a93602e4254ea728c9c2cac9921/pkg/daemon/drain.go) you moved the drain logic into its own controller. I have no idea why I did that! 😀